File: chapter3Python/validating_user_input.py
passes = 1
failures = 0
counter = 0

for students in range(1, 11):
    result = int(input('Enter a number(pass = 1, fail = 2): '))

    while result != 1 and result != 2:
        print('Invalid input')
        result = int(input('Enter a number(pass = 1, fail = 2): '))

    if result == passes and result == 1:
        counter += 1
        print("Passes: ", counter)

    else:
        failures += 1


print("Failures: ", failures)

File: chapter4/ValidatingUserInput.java
package chapter4;

import java.util.Scanner;

public class ValidatingUserInput {
    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        int pass = 0;
        int fail = 0;
        int studentCounter = 1;

        while (studentCounter <= 10) {

            System.out.print("Enter result(1 = pass, 2 = fail): ");
            int result = input.nextInt();

            if (result == 1) {
                pass += 1;
            } else if (result == 2) {
                fail += 1;
            } else {
                System.out.print("Enter a valid result : ");
                continue;
            }
            studentCounter += 1;
        }
        System.out.println("Number of passes " + pass);
        System.out.println("Number of fails " + fail);
        if (pass > 8){
            System.out.println("Bonus To The Instructor!");
        }
      }

    }